src/ReactiveSocket.cpp

```diff
@@ -75,6 +76,20 @@ void ReactiveSocket::requestSubscription(
   automaton->start();
 }
 
+void ReactiveSocket::requestFireAndForget(
```

Comment:
I don't see any value in that yet-another-layer of laziness (that's why I stripped it from the API). If someone wants to be N-times lazy -- lambdas are their friend. We can later on provide a higher-level api, which has no control flow inversion and is as lazy as needed, at a cost of many allocations, indirections, complicated memory management, etc. The lazy API can be built on top of the eager one, stacking in the opposite direction means we always have to pay the cost.

@yschimke ?